charts/spire/charts/spire-server/values.yaml

```diff
@@ -185,12 +185,19 @@ upstreamAuthority:
     enabled: false
     rbac:
       create: true
-    issuer_name: "spire-ca"
+    issuer_name: "spire-server-ca"
     issuer_kind: "Issuer"
     issuer_group: "cert-manager.io"
     # -- Specify to use a namespace other then the one the chart is installed into
     namespace: ""
     kube_config_file: ""
+
+    # -- Creates a Cert-manager CA
+    createCA: false
+    signed:
```

Comment:
This is configuration for the self signed issuer... 

Maybe selfSignedIssuer?

Reply:
The idea is you can swap this out for PKI if desired. It covers the entire spec: field of signing issuer

Reply:
Its the config for the self signed issuer used for building the root CA though. If the user were to do their own, they would need different settings?

Thinking if this setting is configurable, maybe some of the other settings should be too?

Reply:
But you can't currently swap out selfSigned, as you can't remove a key in a helm values file and its hardcoded in values.

These things look potentially useful to configure too?:
https://github.com/spiffe/helm-charts/pull/342/files#diff-c63b6ac3d0fed523e732a68f94aa4bc6ffc1b54e2fe15f2c9aaf8c8d7ecd5eedR25-R27

Reply:
The rub here is that we want a usable set of good defaults. Users could turn off createCA to customize the issuer here. You can actually delete keys, but not in subcharts like used in this project. This style of design isn't very common. It would make more sense to use subfolders in the templates directory instead of several subcharts.

delete a default values: https://github.com/helm/helm/pull/2648/files
subchart delete bug: https://github.com/helm/helm/issues/5184

Reply:
Yeah, its not very consistent. So its best to consider it not usable. I do agree, we need a way to make sure it works out of the box. But if its not unstable, its basically the same thing as hard codinging it but we give the user the false impression that its configurable...

What about a restructuring like:
```yaml
ca:
  create: false
  # -- Set if you don't want a self signed CA. Settings from <pointer to docs>
  rawIssuerSpec: {}
```

Then it does the right thing if the user sets ca.create, and can override the settings to be not self signed if they want?
then in the issuer, condition on the length of rawIssuerSpec?

Reply:
I'm not following here, can you add a complete example? I'm fairly sure `spec: {}` would fail issuer mutating webhook.

Reply:
I moved selfSigned into the template and check for values.yaml override. That should cover your concerns about unsettable defaults.

Reply:
Looks like you were going down a similar path here:
https://github.com/spiffe/helm-charts/pull/342/files#diff-c63b6ac3d0fed523e732a68f94aa4bc6ffc1b54e2fe15f2c9aaf8c8d7ecd5eedR14-R18
with:
```
{{- if .signed.spec }}
  {{- toYaml .signed.spec | nindent 4 }}
{{- else }}
  selfSigned: {}
{{- end }}
```

But its showing up as:
```
| upstreamAuthority.certManager.signed.spec | string | `nil` | See https://cert-manager.io/docs/configuration/ca/ and other options to configure a CA. By default this chart will deploy a selfsigned CA. |
```
in the docs. So its unclear it shouildn't be a string, and that it is actually a dictionary.

if you did it as:
```
{{- if gt (len .signed.spec) 0 }}
  {{- toYaml .signed.spec | nindent 4 }}
{{- else }}
  selfSigned: {}
{{- end }}
```

Then you can define it as ```spec: {}``` in values as a proper dictionary and it would still do the right thing.

The other bit though, its its unclear what you would put in that section, as its only labeled signed.spec but its unclear its options of an Issuer.Spec and not a Certificate.Spec without reading the code. So some kind of restructuring or renaming of the value to make it more clear what the kind of spec is.

Reply:
Changed it to an object.